### PR TITLE
Genetic algo

### PR DESCRIPTION
**Pull request by jameswolfeoliver**
### Done
- implement genetic search classes
- fixed point utils
- Candidate generator
  - randomized initial population 
  - n point arithmetic crossover
  - gaussian mutation
- Search
  - Basic search with max iterations
- Add tests to candidate generator
- Wire up to cirkopt script

### CODE REVIEW

File: python/.pylintrc [hunk 1]
```diff
@@ -18,6 +18,7 @@ disable=missing-module-docstring,
         too-few-public-methods,
         invalid-name,
         fixme,
+        duplicate-code,
```
Comment (thrile) [inline python/.pylintrc:21]: Why is this needed?

**Review by thrile — CHANGES_REQUESTED** [review]
Haven't tested yet, will return to this. I made a couple naming suggestions.

File: python/src/fixed_point.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if precision <= 0:
+        raise Exception("Precision cannot be 0")
+
+    if rounding == Rounding.HALF_UP:
+        return round(val / precision)
+
+    if rounding == Rounding.DOWN:
+        return floor(val / precision)
+
+    return ceil(val / precision)
```
[the remark this answers is not in the document]
Comment (jameswolfeoliver) [reply python/src/fixed_point.py:22]: Sure. Ideally python would be smart enough to know which conditionals covered all the enum values but its not

File: python/.pylintrc (hunk 1 above)
Comment (jameswolfeoliver) [reply python/.pylintrc:21]: It thought imports were duplicate code. I find this kind of linting pretty annoying, so disabled it globally

File: python/tests/test_fixed_point.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import unittest
+
+from src.quantize import quantize, scale, Rounding
+
+
+class TestFixedPoint(unittest.TestCase):
```
Comment (thrile) [inline python/tests/test_fixed_point.py:6]: The file and test name should probably also be updated
